### PR TITLE
feat(api): Add option to fetch Organization details without Projects and Teams

### PR DESCRIPTION
**Pull request by billyvg**
This adds an option to fetch Organization details without projects and teams. This is direction we want to move towards, but we do not want to break the existing details endpoint

Fixes SEN-389

### CODE REVIEW

**Comment by billyvg** [comment]
Thoughts on something like this @wedamija, @markstory ?

File: src/sentry/api/endpoints/organization_details.py [hunk 1]
```diff
@@ -317,10 +317,12 @@ def get(self, request, organization):
                                           team should be created for.
         :auth: required
         """
+        is_skinny = request.GET.get('skinny') == '1'
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_details.py]: I would use `light`, `fast` over `skinny`. :bike:

**Review by markstory — COMMENTED** [review]
I think having an opt-in parameter for the endpoint is a good way to go about this. It will allow us to incrementally adopt this behavior where it will help without impacting existing use-cases.

File: src/sentry/api/endpoints/organization_details.py (hunk 1 above)
Comment (billyvg) [reply src/sentry/api/endpoints/organization_details.py]: 👍 	actually changed it on another branch, do you know if we have any prior nomenclature for this?

Comment (markstory) [reply src/sentry/api/endpoints/organization_details.py]: I know we have a `turbo`. We use `full` or `detailed` in a few places, but I'm not able to find may references to the opposite of `full`.

Comment (wedamija) [reply src/sentry/api/endpoints/organization_details.py]: Should we just invert it?

`is_detailed = request.GET.get('detailed', '1')`

File: src/sentry/api/serializers/models/organization.py [hunk 2]
```diff
@@ -116,43 +116,13 @@ def serialize(self, obj, attrs, user):
         }
 
 
-class DetailedOrganizationSerializer(OrganizationSerializer):
+# Does not include project/teams list
+class LightDetailedOrganizationSerializer(OrganizationSerializer):
```
Comment (wedamija) [inline src/sentry/api/serializers/models/organization.py]: I'd almost rather call this `DetailedOrganizationSerializer` and rename the other one to `DetailedOrganizationSerializerWithTeams` (or something like `WithRelations` or w/e)

Comment (billyvg) [reply src/sentry/api/serializers/models/organization.py]: That would match the naming of our `Team` serializer

File: src/sentry/api/serializers/models/organization.py [hunk 3]
```diff
@@ -205,11 +175,61 @@ def serialize(self, obj, attrs, user, access):
...
-        context['projects'] = serialize(project_list, user, ProjectSummarySerializer())
         context['access'] = access.scopes
         context['pendingAccessRequests'] = OrganizationAccessRequest.objects.filter(
             team__organization=obj,
         ).count()
         context['onboardingTasks'] = serialize(onboarding_tasks, user, OnboardingTasksSerializer())
         return context
+
+
+class DetailedOrganizationSerializerWithProjectsAndTeams(DetailedOrganizationSerializer):
```
Comment (billyvg) [inline src/sentry/api/serializers/models/organization.py]: This is a bit wordy, but it's explicit?

Comment (wedamija) [reply src/sentry/api/serializers/models/organization.py]: Yeah, I can't think of any way to make it more concise. But imo it's better than `LightDetailed...`

File: src/sentry/api/serializers/models/organization.py [hunk 4]
```diff
@@ -116,43 +116,13 @@ def serialize(self, obj, attrs, user):
         }
 
 
+# Does not include project/teams list
```
Comment (wedamija) [inline src/sentry/api/serializers/models/organization.py]: Nit: Probably unnecessary now?

**Review by wedamija — APPROVED** [review]

File: src/sentry/api/endpoints/organization_details.py (hunk 1 above)
Comment (markstory) [reply src/sentry/api/endpoints/organization_details.py]: Wouldn't that require all existing API consumers to change their parameters? The default mode should be backwards compatible imo.

Comment (billyvg) [reply src/sentry/api/endpoints/organization_details.py]: It’s light only when detailed is explicitly 0
